frontend/src/tests/utils/timers.test-utils.ts

```
@@ -0,0 +1,20 @@
+import { nonNullish } from "@dfinity/utils";
+
+const originalTimeout = setTimeout;
+
+export const runResolvedPromises = () =>
+  new Promise((resolve) => originalTimeout(resolve, 0));
+
+// If `millis` is passed, advance time by that much.
+// Otherwise advance time enough to run all current timers.
+// Execute all resolved promises before and after.
+export const advanceTime = async (millis?: number): Promise<void> => {
+  // Make sure the timers are set before we advance time.
```

Not for this PR, could we check this?

Maybe raise an error if timers are not mocked

---

I tried that but jest doesn't expose that information. But it does do its own check and outputs this to the console:
```
    console.warn
      A function to advance timers was called but the timers APIs are 
not replaced with fake timers. Call `jest.useFakeTimers()` in this 
test file or enable fake timers for all tests by setting 'fakeTimers': 
{'enableGlobally': true} in Jest configuration file.
      Stack Trace:
```

But that's actually not what I meant by this comment.
What I meant is that we call `await runResolvedPromises();` to make sure that our code under test calls `setTimeout` (rather than blocked on awaits) before we call `jest.runOnlyPendingTimers()` or `jest.advanceTimersByTime(millis);`.

---

aaahh, ok, got it. Thanks for the explanation